pyanaconda/modules/payloads/payload/dnf/dnf.py

```diff
@@ -335,12 +335,7 @@ def get_repo_configurations(self):
         :return: RepoConfiguration structures for attached sources.
         :rtype: RepoConfigurationData instances
         """
-        structures = []
-
-        for source in self.sources:
-            structures.append(source.generate_repo_configuration())
-
-        return structures
+        return list(filter(None, [s.generate_repo_configuration() for s in self.sources]))
```

Comment:
The function can return None now if the source is not set up. It shouldn't happen, but I hit this in unit tests. Anyway, the logic around `generate_repo_configuration` will go away, because it will be eventually replaced with the new `repository` property. It is just temporary.